locust_tasks/locustfile.py

```diff
@@ -166,49 +189,163 @@ def correct_address(self):
             'request-mobile-number': '07714 330 933'
-        })
+        }, catch_response=True) as response:
+            verify_response('RequestUacSms-EnterMobileNumber', self, response, 200, Page.CONFIRM_MOBILE, "933")
 
-    @task(6)
+    @task(7)
     def confirm_mobile_number(self):
-        self.client.post("/request-access-code/confirm-mobile", {
+        """
+        POST 'yes' to confirm mobile number
+        """
+        with self.client.post("/en/requests/access-code/confirm-mobile/", {
             'request-mobile-confirmation': 'yes'
-        })
+        }, catch_response=True) as response:
+            verify_response('RequestUacSms-ConfirmMobileNumber', self, response, 200, Page.CODE_SENT)
+
+
+class RequestNewCodePost(SequentialTaskSet):
+
+    # All users arrive at the start page
+    @task(1)
+    def start_page(self):
+        """
+        GET Start page
+        """
+        self.case = get_next_case()
+        with self.client.get('/en/start/', catch_response=True) as response:
+            verify_response('RequestUacPost-Start', self, response, 200, Page.START)
+
+    @task(2)
+    def enter_postcode(self):
+        """
+        POST postcode
+        """
+        with self.client.post("/en/requests/access-code/enter-address/", {
+            'form-enter-address-postcode': self.case['postcode']
+        }, catch_response=True) as response:
+            self.address_to_select = extractAddress(response, self.case["uprn"])
+            verify_response('RequestUacPost-EnterAddress', self, response, 200, Page.SELECT_ADDRESS,
+                            self.case["postcode"])
+
+    @task(3)
+    def select_address(self):
+        """
+        POST uprn and whole address extracted as JSON from the HTML in previous task
+        """
+        logger.info("Address selected: " + self.address_to_select)
+        with self.client.post("/en/requests/access-code/select-address/", {
+            'form-select-address': self.address_to_select
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-SelectAddress', self, response, 200, Page.ADDRESS_CORRECT, self.case["addressLine1"])
+
+    @task(4)
+    def confirm_address(self):
+        """
+        POST 'yes' to confirm address
+        """
+        with self.client.post("/en/requests/access-code/confirm-address/", {
+            'form-confirm-address': 'yes'
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-ConfirmAddress', self, response, 200, Page.SELECT_METHOD, "Post")
+
+    @task(5)
+    def select_method(self):
+        """
+        POST 'post' to select post as method of sending UACs
+        """
+        with self.client.post("/en/requests/access-code/select-method/", {
+            'form-select-method': 'post'
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-SelectMethod', self, response, 200, Page.ENTER_NAME, "")
+
+    @task(6)
```

Comment:
All instances of `@task(<num>)` need to be replaced with `@task`.
This is because of the way that locust handles the weighting parameter. Although it steps through the tasks one by one, it's executing the task multiple times. 
For example the previous  task (select_method) is executed 5 times, this one 6 and confirm_name_address 7 times.
Don't think this is what the Locust docs say, but it is what's happening!
This is very obvious if you run with just a single user. The number of requests for each endpoint very quickly becomes unbalanced, with the count for confirm_name_address being 7 approx. 7 times higher than the first task.
Remove the parameter and locust executes each task just once.

Reply:
Oh! Well spotted - that explains why it wasn't going through the test data as quickly as I had expected it to. I shall definitely correct that.

Reply:
done